aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,598 @@
+                - 'pose': current pose of the aria in CPF frame i.e. ariaWorld_T_cpf as Sophus SE3
+                - 'device_pose': current pose of the aria in device frame i.e. ariaWorld_T_device as Sophus SE3
+                - 'ros_pose': current pose of the aria in CPF frame i.e. ariaWorld_T_cpf as ROS Pose
+                - 'timestamp': the ros timestamp when frame dict was created (i.e. when this function was called)
+            outputs (dict): Dictionary of outputs from the april tag detector
+            detect_qr (bool, optional): Whether to detect april tag. Defaults to False.
+            detect_objects (bool, optional): Whether to detect objects. Defaults to False.
+            object_label (str, optional): Object label to detect. Defaults to "milk bottle".
+
+        Publishes:
+            - static tf: ROS TransformStamped message for marker frame wrt ariaWorld
+            - /aria_pose_of_interest: ROS PoseStamped message for ariaWorld_T_cpf at pose of interest i.e. pose where it detected the object
+            - /aria_pose_of_interest_marker: ROS Marker message for ariaWorld_T_cpf at pose of interest i.e. pose where it detected the object
+
+        Returns:
+            Tuple[np.ndarray, dict]: Tuple of:
+                - viz_img (np.ndarray): Image with april tag and object in hand detections
+                - outputs (dict): Dictionary of outputs from the april tag detector
+        """
+        # Initialize camera_T_marker to None & object_scores to empty dict for current image frame
+        camera_T_marker = None
+        object_scores = {}
+
+        viz_img = frame.get("image")
+        if detect_qr:
+            (viz_img, camera_T_marker) = self.april_tag_detector.process_frame(img_frame=frame.get("raw_image"))  # type: ignore
+            # Rotate current image frame
+            viz_img = self._rotate_img(img=viz_img)
+
+        if camera_T_marker is not None:
+            device_T_marker = self.device_T_camera * camera_T_marker
+            ariaWorld_T_device = frame.get("device_pose")
+
+            # Frame: world = ariaWorld
+            # Frame: a = device (aria's device / left slam camera frame)
+            # Frame: b = marker / qr code
+            (
+                self.marker_positions_list,
+                self.marker_quaternion_list,
+                self.avg_ariaWorld_T_marker,
+            ) = get_running_avg_world_T_b(
+                current_avg_world_T_b=self.avg_ariaWorld_T_marker,
+                world_T_b_position_list=self.marker_positions_list,
+                world_T_b_quaternion_list=self.marker_quaternion_list,
+                world_T_a=ariaWorld_T_device,
+                a_T_b=device_T_marker,
+            )
+
+            if self.avg_ariaWorld_T_marker is not None:
+                avg_marker_T_ariaWorld = self.avg_ariaWorld_T_marker.inverse()
+
+                # Publish marker pose in ariaWorld frame
+                self.static_tf_broadcaster.sendTransform(
```

Comment:
should this be published every time this function is called, or only the first time the QR is detected?

Basically, should `if camera_T_marker is not None:` be within the previous `if detect_qr:` loop?

Reply:
We need to publish this per cycle otherwise the rviz marker will fade due to old timestamp.
That said, `if camera_T_marker is not None:` can be within previous condition-set as we do not have any alt way of getting `camera_T_marker`

Reply:
As @zephirefaith  mentioned
1. The TF that we publish here gets registered to the TF tree which contains (spotWorld->spot->marker). This current TF publishes (marker->ariaWorld). Speaking about marker->ariaWorld, many-a-times, due to noise, ariaWorld might be shifted from where it actually is. If user keeps looking at marker for a few seconds steadily, this noise gets reduced and that (marker->ariaWorld) detections are more "acceptable". So with that idea, it is good and useful to keep publishing (marker->ariaWorld) transformation for making sure the transform is relatively noise-free.
2. It would make no difference if we move the logic from within `if camera_T_marker is not None:` to inside `if detect_qr:`. If detect_qr bool is false, it is just going to get by-passed all the time. Logic will never enter there so we should be good.